src/controllers/scripting/legacy/controllerscriptinterfacelegacy.cpp

```diff
@@ -31,7 +31,9 @@ constexpr double kBrakeRampToRate = 0.01;
 ControllerScriptInterfaceLegacy::ControllerScriptInterfaceLegacy(
         ControllerScriptEngineLegacy* m_pEngine, const RuntimeLoggingCategory& logger)
         : m_pScriptEngineLegacy(m_pEngine),
-          m_logger(logger) {
+          m_logger(logger),
+          m_pPlayerManagerProxy(mixxx::qml::QmlPlayerManagerProxy::create(m_pEngine->jsEngine().get(), m_pEngine))
```

Comment:
Still struggling with C++'s pointers. Not sure about `m_pEngine->jsEngine().get()` here. I feel like ideally I should be using `.get()` which seem to defeat `std::shared_ptr`. I though I could use C++ smart pointers (`shared_ptr` and `unique_ptr`) as if they were raw pointers but no.

Reply:
yeah, I see how thats confusing. In this case its fine though because `create` does not take the pointer for storing (in fact, its unused, we should double check that). So reaching into the shared_ptr and taking the raw one is completely fine in this case...